src/FieldEncryptProcessEntities.php

```diff
@@ -338,4 +338,55 @@ public function updateStoredField($field_name, $field_entity_type, $original_enc
     }
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function entitySetCacheTags(ContentEntityInterface $entity, &$build) {
+    $cache_exclude_fields = $this->getCacheExcludeFields($entity);
+    foreach ($cache_exclude_fields as $field_name) {
+      $build[$field_name]['#cache']['max-age'] = 0;
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function entityCheckPersistentCache(ContentEntityInterface $entity) {
+    $cache_exclude_fields = $this->getCacheExcludeFields($entity);
+    if (!empty($cache_exclude_fields)) {
+      // Some fields don't want to be cached, so clear the persistent entity
+      // cache, to avoid their values to be cached unencrypted.
+      /** @var \Drupal\Core\Entity\ContentEntityStorageInterface $storage */
+      $storage = $this->entityManager->getStorage($entity->getEntityTypeId());
+      $storage->resetCache([$entity->id()]);
```

Comment:
This does NOT prevent this entity from being cached, it just means it's first stored in cache and then removed. That misses the point. You probably need to override the classes doing this caching in the first place and preventing those from being cached at all.

Or, better yet…

- `\Drupal\Core\Entity\EntityType::$static_cache`
- `\Drupal\Core\Entity\EntityType::$render_cache`
- `\Drupal\Core\Entity\EntityType::$persistent_cache`

I think the better approach would be to override this metadata for entity types that have any fields that have `field_encrypt` enabled. That would prevent them from ever being cached.